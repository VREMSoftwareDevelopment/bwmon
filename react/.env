VITE_VERSION=$npm_package_version
VITE_NAME=$npm_package_name
VITE_DESCRIPTION=$npm_package_description
