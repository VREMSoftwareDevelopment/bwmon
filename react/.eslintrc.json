{
    "extends": [
        "eslint:recommended",
        "plugin:import/errors",
        "plugin:react/recommended",
        "plugin:jsx-a11y/recommended",
        "plugin:prettier/recommended"
    ],
    "plugins": ["react", "import", "jsx-a11y", "jest", "prettier"],
    "settings": {
        "react": {
            "version": "detect"
        }
    },
    "rules": {
        "react/prop-types": "off",
        "indent": ["error", 4],
        "linebreak-style": "error",
        "quotes": ["error", "single"],
        "no-unused-vars": [
            "error",
            {
                "vars": "all",
                "varsIgnorePattern": "^_",
                "args": "after-used",
                "argsIgnorePattern": "^_",
                "caughtErrors": "all",
                "caughtErrorsIgnorePattern": "^_"
            }
        ]
    },
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "env": {
        "es6": true,
        "browser": true,
        "node": true,
        "jest/globals": true
    }
}
