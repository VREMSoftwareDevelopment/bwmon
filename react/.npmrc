save=true
puppeteer_skip_download=true
PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true