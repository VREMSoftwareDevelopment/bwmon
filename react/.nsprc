{
    "1070415": {
        "active": true,
        "expiry": "2022-12-31",
        "notes": "Ignored - waiting for update"
    }
}
