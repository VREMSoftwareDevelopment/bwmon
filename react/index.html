<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <link rel="icon" href="/favicon.ico" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
        <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
        <meta name="theme-color" content="#000000" />
        <meta name="Bandwidth Monitor" content="Bandwidth MonitorApplication" />
        <link rel="manifest" href="/manifest.json" />
        <base href="/" />
        <title>Bandwidth Monitor</title>
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
        <script type="module" src="/src/index.jsx"></script>
    </body>
</html>
