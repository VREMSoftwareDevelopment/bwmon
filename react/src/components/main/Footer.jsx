/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@mui/material';
import { footerStyles } from '../../utils/StylesUtils';

const Footer = ({ currentTime }) => {
    const classes = footerStyles();

    return (
        <div className={classes.root}>
            <Typography data-testid="app-footer1" id="app-footer1" variant="subtitle2">
                All usage information is in gigabytes
            </Typography>
            <Typography data-testid="app-footer2" id="app-footer2" variant="subtitle2">
                This page was generated on {currentTime}
            </Typography>
        </div>
    );
};

// Stryker disable next-line all
Footer.propTypes = { currentTime: PropTypes.string.isRequired };

export default Footer;
