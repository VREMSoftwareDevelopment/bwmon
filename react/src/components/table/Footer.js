/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { TableHead, TableRow, TableCell } from '@mui/material';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" data-testid={prefix + '-footer'} id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                const text =
                    index === 0
                        ? 'Totals'
                        : cellInfo.footer
                          ? cellInfo.convert
                              ? cellInfo.convert(values[cellInfo.id])
                              : values[cellInfo.id]
                          : '';
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

export default Footer;
