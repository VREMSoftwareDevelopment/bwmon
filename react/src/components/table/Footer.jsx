/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { TableHead, TableRow, TableCell } from '@mui/material';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" data-testid={prefix + '-footer'} id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                let text = '';
                if (index === 0) {
                    text = 'Totals';
                } else if (cellInfo.footer) {
                    text = cellInfo.convert ? cellInfo.convert(values[cellInfo.id]) : values[cellInfo.id];
                }
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

Footer.propTypes = {
    prefix: PropTypes.string.isRequired,
    cellInfos: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            align: PropTypes.string,
            footer: PropTypes.bool,
            convert: PropTypes.func,
        })
    ).isRequired,
    values: PropTypes.object.isRequired,
};

export default Footer;
