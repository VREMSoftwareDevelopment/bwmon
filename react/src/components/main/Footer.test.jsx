/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import Footer from './Footer';

describe('Footer', () => {
    const theme = createTheme();

    const renderComponent = (props) =>
        render(
            <ThemeProvider theme={theme}>
                <Footer {...props} />
            </ThemeProvider>
        );

    it('renders static text and current time', () => {
        const currentTime = 'Sunday, 15 December 2024, 13:15';
        renderComponent({ currentTime: currentTime });
        expect(screen.getByTestId('app-footer1')).toHaveTextContent('All usage information is in gigabytes');
        expect(screen.getByTestId('app-footer2')).toHaveTextContent(`This page was generated on ${currentTime}`);
    });
});
