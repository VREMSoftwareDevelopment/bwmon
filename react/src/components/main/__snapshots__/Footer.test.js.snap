// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    All usage information is in gigabytes
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    This page was generated on 
    October 20, 2020, 11:25:35 AM EDT
  </h6>
</div>
`;
